### PR TITLE
qa: Relax so that the subscriber is ready before publishing zmq messages

### PR DESCRIPTION
**Pull request by promag**
Prevents the syndrome "slow joiner" - see http://zguide.zeromq.org/py:all#sockets-and-patterns - by relaxing before publishing messages.

### CODE REVIEW

**Comment by promag** [comment]
Follow up to https://github.com/bitcoin/bitcoin/pull/16404#issuecomment-525449503.

File: test/functional/interface_zmq.py [hunk 1]
```diff
@@ -71,6 +71,9 @@ def test_basic(self):
 
         self.restart_node(0, ["-zmqpub%s=%s" % (sub.topic.decode(), address) for sub in [hashblock, hashtx, rawblock, rawtx]])
         connect_nodes(self.nodes[0], 1)
+        socket.connect(address)
+        # Relax so that the subscriber is ready before publishing zmq messages
+        sleep(1)
```
[the remark this answers is not in the document]
Comment (promag) [reply test/functional/interface_zmq.py]: I don't think so, I'll lower it.

[the remark this answers is not in the document]
Comment (promag) [reply test/functional/interface_zmq.py]: Done.

**Comment by emilengler** [comment]
concept ACK but I would prefer a smaller amount than 0.2 because humans can detegt a delay there. 0.02 would be better I think

**Comment by promag** [comment]
@MarcoFalke let me know if you see the test fail again.

**Comment by dongcarl** [comment]
@promag is this error the one that is supposed to be fixed by this? https://cirrus-ci.com/task/4605167582576640?command=ci#L4905

**Comment by MarcoFalke** [comment]
@dongcarl See 

#20618
#20590
#20538

**Pull request merged**